packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1507,6 +1508,46 @@ function isAtlasStreamsInstance(
   }
 }
 
+const NonRetryableErrorCodes = [3000, 3003, 4004, 1008] as const;
+const NonRetryableErrorDescriptionFallbacks: {
+  [code in typeof NonRetryableErrorCodes[number]]: string;
+} = {
+  3000: 'Unauthorized',
+  3003: 'Forbidden',
+  4004: 'Not Found',
+  1008: 'Violated policy',
+};
+
+function isNonRetryableHeartbeatFailure(evt: ServerHeartbeatFailedEvent) {
+  return NonRetryableErrorCodes.some((code) =>
+    evt.failure.message.includes(`code: ${code},`)
+  );
+}
+
+function getDescriptionForNonRetryableError(error: Error): string {
+  // Give a description from the error message when provided, otherwise fallback
+  // to the generic error description.
+  const reason = error.message.match(/code: \d+, reason: (.*)$/)?.[1];
+  return reason && reason.length > 0
+    ? reason
+    : NonRetryableErrorDescriptionFallbacks[
+        Number(
+          error.message.match(/code: (\d+),/)?.[1]
+        ) as typeof NonRetryableErrorCodes[number]
+      ] ?? 'Unknown';
+}
+
+const openConnectionClosedWithNonRetryableErrorToast = (
+  connectionInfo: ConnectionInfo,
+  error: Error
+) => {
+  openToast(`non-retryable-error-encountered--${connectionInfo.id}`, {
+    title: `Unable to connect to ${getConnectionTitle(connectionInfo)}`,
+    description: `Reason: ${getDescriptionForNonRetryableError(error)}`,
```

Comment:
We can mention here that a user will have to refresh or disconnect and reconnect before they can use the connection.

Reply:
<img width="414" alt="Screenshot 2025-01-13 at 5 14 31 PM" src="https://github.com/user-attachments/assets/18026314-0bfd-4f19-aa2e-5f4faf89b83a" />